src/commcare_cloud/ansible/roles/kafka/templates/server.properties.j2

```diff
@@ -117,3 +117,9 @@ zookeeper.connect={% for host in groups['zookeeper'] %}{{ host }}:{{ zookeeper_c
 
 # Timeout in ms for connecting to zookeeper
 zookeeper.connection.timeout.ms=1000000
+{% if kafka_version == "2.4.1" %}
+inter.broker.protocol.version=2.4
+log.message.format.version=2.4
+listeners=PLAINTEXT://:9092
+advertised_listener=PLAINTEXT://{{inventory_hostname}}:9092
+{% endif %}
```

Comment:
can we do a similar thing for `advertised.host.name={{ inventory_hostname }}` on line [34](https://github.com/dimagi/commcare-cloud/pull/3786/files#diff-c4593a9a7d190dc9851e42057de19cdeR34) (only check for the older version).

Also, are there any other legacy settings that should be removed?